File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

class Main{
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int V = Integer.parseInt(st.nextToken());

        List<List<Integer>> graph = new ArrayList<>();
        for(int i=0; i<=N; i++){
            graph.add(new ArrayList<>());
        }
        for(int i=0; i<M; i++){
            st = new StringTokenizer(br.readLine());
            int u = Integer.parseInt(st.nextToken());
            int v = Integer.parseInt(st.nextToken());

            graph.get(u).add(v);
            graph.get(v).add(u);
        }
        br.close();

        // 각 노드의 인접 리스트를 정렬
        for (int i = 1; i <= N; i++) {
            Collections.sort(graph.get(i));
        }

        boolean[] visitied = new boolean[N+1];
        StringBuilder dfsResult = new StringBuilder();
        dfs(graph, V, visitied, dfsResult);
        System.out.println(dfsResult.toString().trim());

        String bfsResult = bfs(graph, V, N);
        System.out.println(bfsResult.trim());
    }

    private static void dfs(List<List<Integer>> graph, int current, boolean[] visitied, StringBuilder result){
        visitied[current] = true;
        result.append(current).append(" ");

        for(int next : graph.get(current)){
            if(!visitied[next]){
                dfs(graph, next, visitied, result);
            }
        }
    }

    private static String bfs(List<List<Integer>> graph, int start, int nodes){
        Queue<Integer> que = new LinkedList<>();
        boolean[] visited = new boolean[nodes+1];
        StringBuilder result = new StringBuilder();

        que.add(start);
        visited[start] = true;

        while(!que.isEmpty()){
            int current = que.poll();
            result.append(current).append(" ");

            for(int next : graph.get(current)){
                if(!visited[next]){
                    que.add(next);
                    visited[next] = true;
                }
            }
        }
        return result.toString();
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
const input = require('fs').readFileSync("/dev/stdin").toString().trim().split('\n');
const [vertices, edges, start] = input[0].split(" ").map(Number);
const graph = Array.from({ length: vertices + 1 }, () => []);

// 간선 입력
for (let i = 1; i <= edges; i++) {
    const [u, v] = input[i].split(" ").map(Number);
    graph[u].push(v);
    graph[v].push(u);
}

// 각 노드의 인접 리스트 정렬
for (let i = 1; i <= vertices; i++) {
    graph[i].sort((a, b) => a - b);
}

const dfsVisited = Array(vertices + 1).fill(false);
const dfsResult = [];
dfs(graph, start, dfsVisited, dfsResult);
console.log(dfsResult.join(" "));

const bfsResult = bfs(graph, start, vertices);
console.log(bfsResult.join(" "));

function dfs(graph, current, visited, result) {
    visited[current] = true;
    result.push(current);

    for (const next of graph[current]) { // 수정된 문법
        if (!visited[next]) {
            dfs(graph, next, visited, result);
        }
    }
}

function bfs(graph, start, vertices) {
    const result = [];
    const que = [];
    const visited = Array(vertices + 1).fill(false);

    visited[start] = true;
    que.push(start);

    while (que.length > 0) {
        const current = que.shift();
        result.push(current);

        for (const next of graph[current]) { // 수정된 문법
            if (!visited[next]) {
                que.push(next);
                visited[next] = true;
            }
        }
    }
    return result;
}